crossfire/clients/occurrences.py

```diff
@@ -164,3 +179,58 @@ def __call__(self):
             return GeoDataFrame(self.data)
 
         return self.data
+
+
+def _flatten_df(data, nested_columns):
+    def _flatten_col(row, column_name):
+        column_data = row[column_name]
+        return Series(
+            {
+                f"{column_name}_{key}": value
+                for key, value in column_data.items()
+            }
+        )
+
+    keys = set(data.columns) & nested_columns
+    for key in keys:
+        data = concat(
+            [
+                data.drop(key, axis=1),
+                data.apply(_flatten_col, args=(key,), axis=1),
+            ],
+            axis=1,
+        )
+    return data
+
+
+def _flatten_list(data, nested_columns):
+    keys = set(data[0].keys()) & nested_columns
+    for item in data:
+        for key in keys:
+            if key in item.keys():
+                if item.get(key):
+                    item.update(
+                        {f"{key}_{k}": v for k, v in item.get(key).items() if v}
+                    )
+                    item.pop(key)
```

Comment:
These lines have four lookups for the keys in the dictionary. This is unnecessary.

But first, why `.keys()` if the lookup is already in the keys?

```suggestion
            if key in item:
                if item.get(key):
                    item.update(
                        {f"{key}_{k}": v for k, v in item.get(key).items() if v}
                    )
                    item.pop(key)
```

Now, let's save the result of the first lookup and re-use instead of performing another lookup:

```suggestion
            if key in item:
                value = item.get(key)
                if value:
                    item.update(
                        {f"{key}_{k}": v for k, v in value.items() if v}
                    )
                    item.pop(key)
```

Now, let's get the result of the `.pop` instead of another lookup to remove the item:

```suggestion
            if key in item:
                value = item.pop(key)
                if value:
                    item.update(
                        {f"{key}_{k}": v for k, v in value.items() if v}
                    )
```